### PR TITLE
Fix ping method invoked before Init method for pluggable components

### PR DESCRIPTION
**Pull request by mcandeia**
Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

# Description

Using `WithBlock` grpc option to block connection until it is ready to be used rather than relying on `WaitForReady` for `Ping` method call when initializing pluggable components gRPC connection.

See the related issue for the detailed explanation.
## Issue reference

Please reference the issue this PR will close: #5658 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/components/pluggable/grpc.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ func (g *GRPCConnector[TClient]) Dial(name string) error {
 
 	g.Client = g.clientFactory(grpcConn)
 
-	return g.Ping()
+	return nil
```
Comment (mukundansundar) [inline pkg/components/pluggable/grpc.go:103]: @mcandeia Why can't we still return the value from g.Ping()?

**Review by mukundansundar — DISMISSED** [review]
lgtm

**Review by artursouza — APPROVED** [review]

**Pull request merged**